### PR TITLE
Crypto module - taking from Eitan's work

### PR DESCRIPTION
**Pull request by rabbitonweb**

### CODE REVIEW

**Comment by rabbitonweb** [comment]
@kirkwood @jeremybusk this PR is ready for review

File: scripts/install_sodium.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+set -e
```
Comment (jeremybusk) [inline scripts/install_sodium.sh]: For set command in BASH I'd like to always use this line unless there is a reason not too. I'd like to make this standard. Not a biggie but I think better.
set -euxo pipefail
This is a good overview of why. 
https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/
Maybe less is more but I think the use of set command is reasonable

Comment (rabbitonweb) [reply scripts/install_sodium.sh]: @jeremybusk this is brilliant! thx!

Comment (rabbitonweb) [reply scripts/install_sodium.sh]: alright, added!

**Pull request merged**